### PR TITLE
[PLATFORM-2230]: Create an MVP for webkit support

### PR DESCRIPTION
**Pull request by MaeIsBad**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-2230

this is very unfinished and could use a bunch of refactors but it's good enough for an mvp

### CODE REVIEW

**Review by cpiemontese — COMMENTED** [review]
I can't comment that much on the Rust code itself, the question I have is with maintenability

- We have Nix, which not many people in Prima are familiar with
- We have "complex" Rust code

The second thing could probably be improved with more documentation, the first one I don't know... I don't want to kill the Nix dream but it is something we need to consider

That said, really nice job and really clean code :smile:

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+}
+
+#[cfg(test)]
+mod test {
+    #[test]
+    fn test_add() {
+        assert_eq!(1 + 2, 3);
+    }
+}
```
[the remark this answers is not in the document]
Comment (MaeIsBad) [reply src/lib.rs:82]: yep

**Comment by MaeIsBad** [comment]
> We have Nix, which not many people in Prima are familiar with

yeahhh, but the alternative for our use case was to make a debian package, which I don't know how to do myself, and I don't think many more people at prima know how to.

> We have "complex" Rust code

yeahhhh, this is interacting with GTK so not much can be done in terms of fixing it

**Review by cottinisimone — APPROVED** [review]
For the code part it seems ok to me. I didn't even took a look at the nix part :)

**Review by cpiemontese — APPROVED** [review]
I think it's fine for an MVP, if the Nix or Rust code can be documented better let's do that otherwise it's ok

**Pull request merged**